src/gennylib/include/gennylib/v1/GlobalRateLimiter.hpp

```diff
@@ -84,6 +87,30 @@ class BaseGlobalRateLimiter {
      * appropriate back-off strategy if this function returns false.
      */
     bool consumeIfWithinRate(const typename ClockT::time_point& now) {
+
+        // Logic for percentile rates.
+        bool curFullSpeed = _fullSpeed.load();
+        if (curFullSpeed) {
+            _burstCount++;
+            auto _nsSincePhase = ClockT::now().time_since_epoch().count() - _lastEmptiedTimeNS;
+
+            // 3 iterations or 1 minute, whichever is longer.
+            if (_iters >= _numUsers * 3 && _nsSincePhase >= _nsPerMinute) {
+                const auto success =
+                    _fullSpeed.compare_exchange_weak(curFullSpeed, false);
+                if (!success) {
+                    return true;
+                }
+                // Reconfigure as a "normal" rate limiter running for the first time.
+                _burstSize = _burstCount * _percent.value() / 100;
```

Comment:
I think that would be a bit weird; this is all doing integer division since `burstSize` is an int and to have a subzero fraction in the RateSpec we'd use a double and cast back to an int here, and I think that'd be a bit trickier to test.